### PR TITLE
Add support for 'running out of ammo' notifications in cannon plugin with a custom threshold

### LINKED ISSUE #1820: Make the notification for cannonballs customizable
To be able customize at what point you get a notification that cannon is empty.
e.g "Cannonball notification 1 - 30: '5' " This will give a notificaiton when cannon has 5 cannonballs inside. 

This gives you enough time to fill the cannon, before it stops. Also less clicking.

### PR DESCRIPTION
**Pull request by jpabeem**
ref #1820 
* Add support for 'running out of ammo' notifications with a custom configurable threshold
* Rearrange the settings, now grouped and not cluttered anymore:

Closes #1820

### CODE REVIEW

**Comment by sethtroll** [comment]
Instead of having 2 notifications being sent (one for being low and one for being empty), I would instead just have a config option for when to notify you, and have it set to 0 by default.

**Comment by jpabeem** [comment]
@sethtroll what to do when people want both? I agree with 1 notification being enough though, will improve this when I have the time.

**Comment by jpabeem** [comment]
Thanks for feedback @Kamielvf , it should work as you and others described now.

File: runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonConfig.java [hunk 1]
```diff
@@ -37,29 +37,54 @@
...
+		position = 2
 	)
 	default boolean showInfobox()
 	{
 		return false;
 	}
 
+	@ConfigItem(
+		keyName = "showEmptyCannonNotification",
+		name = "Out of ammo notification",
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonConfig.java]: "Low ammo notification"?

File: runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonPlugin.java [hunk 2]
```diff
@@ -173,6 +177,24 @@ public void checkSpots()
 		}
 	}
 
+	@Schedule(
+			period = 1,
+			unit = ChronoUnit.SECONDS
+	)
+	public void checkAmmo()
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonPlugin.java]: Any reason why this doesn't just check the count whenever the cannonball is fired?

**Comment by deathbeam** [comment]
Closing as this is very old PR now and has a lot of merge conflicts, reopen if you want to re-implement it.

**Pull request closed without merging**